File: SourceCode/SQLAPI/POS.Data/Entities/Expense/Expense.cs
using POS.Data.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace POS.Data
{
    public class Expense : BaseEntity
    {
        public Guid Id { get; set; }
        public string Reference { get; set; }
        public Guid ExpenseCategoryId { get; set; }
        [ForeignKey("ExpenseCategoryId")]
        public ExpenseCategory ExpenseCategory { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }
        public Guid? ExpenseById { get; set; }
        public string Description { get; set; }
        public DateTime ExpenseDate { get; set; }
        public string ReceiptName { get; set; }
        public string ReceiptPath { get; set; }
        [ForeignKey("ExpenseById")]
        public User ExpenseBy { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalTax { get; set; }
        public Guid LocationId { get; set; }
        [ForeignKey("LocationId")]
        public Location  Location { get; set; }
        public List<ExpenseTax> ExpenseTaxes { get; set; }
    }
}

File: SourceCode/Angular/src/app/core/domain-classes/expense.ts
import { ExpenseCategory } from "./expense-category";
import { ExpenseTax } from "./expenseTax";
import { User } from "./user";

export class Expense {
  id!: string;
  reference!: string;
  expenseCategoryId!: string;
  amount!: number;
  expenseById!: string;
  description!: string;
  receiptName!: string;
  expenseDate?: Date;
  expenseBy?: User;
  expenseCategory!: ExpenseCategory;
  expenseTaxIds?: string[];
  expenseTaxes?: ExpenseTax[];
  totalTax?: number;
}
